src/librustc_errors/lib.rs

```diff
@@ -103,6 +104,7 @@ pub trait CodeMapper {
     fn span_to_filename(&self, sp: Span) -> FileName;
     fn merge_spans(&self, sp_lhs: Span, sp_rhs: Span) -> Option<Span>;
     fn call_span_if_macro(&self, sp: Span) -> Span;
+    fn load_source_for_filemap(&self, file: FileName) -> bool;
```

Comment:
Maybe the long term solution would be to move `CodeMapper` to `libsyntax_pos`? Or better yet, move `CodeMap` there and remove it entirely? I'm not sure which other consequences this might have though.